src/python/yaramod_python.cpp

```diff
@@ -538,6 +539,7 @@ void addExpressionClasses(py::module& module)
 		.def("accept", &Expression::accept)
 		.def("get_text", &Expression::getText, py::arg("indent") = std::string{})
 		.def("exchange_tokens", py::overload_cast<Expression*>(&Expression::exchangeTokens))
+		.def_property_readonly("uid", [](const Expression& self) { return self.getUid();} )
```

Comment:
You can directly use `&Expression::getUid`. You have maybe copied it over from the code below but there it's needed because even though we are exposing `text` as a property, the function `getText` has a default argument which Python cannot handle. If you look even more down, you can see that we are many times actually mapping the properties directly to method pointers.

Reply:
Oh ye, thanks for that, fixed